From 3e2c1ddd06be97ba75104b1be4b6fdbd08e16bbe Mon Sep 17 00:00:00 2001
From: Alex Suykov <alex.suykov@gmail.com>
Date: Wed, 2 Jan 2019 20:37:47 +0200
Subject: [PATCH] fix Yocto intermittent failures with modules

Having -Ilib/ above -Idist/... in miniperl_top means miniperl may
attempt to switch from dist/ to lib/ while the modules are being
built, possibly picking incompletely-written files there.

Any module available via -Idist/* should only be loaded from dist/
and never from lib/ even if it gets installed into lib/ at some point.

Upstream-Status: Submitted [https://github.com/arsv/perl-cross/issues/72]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 miniperl_top | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/miniperl_top b/miniperl_top
index 96ab1a2..a0426b9 100755
--- a/miniperl_top
+++ b/miniperl_top
@@ -25,7 +25,6 @@ top=$(cd $top; pwd)
 exec $top/miniperl\
 	-I$top/cnf/stub\
 	-I$top/cnf/cpan\
-	-I$top/lib\
 	-I$top/cpan/AutoLoader/lib\
 	-I$top/dist/Exporter/lib\
 	-I$top/dist/Cwd\
@@ -50,4 +49,5 @@ exec $top/miniperl\
 	-I$top/cpan/parent/lib\
 	-I$top/cpan/version/lib\
 	-I$top/dist/Pod-Simple/lib\
+	-I$top/lib\
 	"$@"
